File: dotnet/samples/SimpleOrm.Sample/Models/TransactionDetail.cs
namespace SimpleOrm.Sample.Models;

/// <summary>
/// Table <c>transaction_details</c> (STRICT). Key: <c>id</c>, database-generated;
/// <c>transaction_id</c> references <c>transactions</c>. The child side of the
/// json_group_array nesting pattern documented at milestone 4 (CLAUDE.md §7.10).
/// </summary>
[Table("transaction_details")]
[Index(nameof(TransactionId))]
public sealed class TransactionDetail : BaseModel
{
    [Key]
    [Generated]
    [Column]
    public long Id { get; set; }

    [Column]
    [ForeignKey(typeof(Transaction))]
    public long TransactionId { get; set; }

    /// <summary>Populated only by the library (Level 2 loading); no public setter, so it can never disagree with <see cref="TransactionId"/>.</summary>
    [ManyToOne(nameof(TransactionId))]
    public Transaction? Transaction { get; private set; }

    [Column]
    public required string Description { get; set; }

    [Column]
    public int Quantity { get; set; }

    [Column]
    public decimal UnitPrice { get; set; }
}

File: php/samples/SimpleOrm.Sample/src/Models/Tables/TransactionDetail.php
<?php

declare(strict_types=1);

namespace SimpleOrm\Sample\Models\Tables;

use SimpleOrm\Metadata\Attributes\Column;
use SimpleOrm\Metadata\Attributes\ForeignKey;
use SimpleOrm\Metadata\Attributes\Generated;
use SimpleOrm\Metadata\Attributes\Index;
use SimpleOrm\Metadata\Attributes\Key;
use SimpleOrm\Metadata\Attributes\ManyToOne;
use SimpleOrm\Metadata\Attributes\Table;
use SimpleOrm\Metadata\ColumnType;
use SimpleOrm\Types\Decimal;

/**
 * Table `transaction_details` (STRICT). Key: `id`, database-generated;
 * `transaction_id` references `transactions`. The child side of the
 * json_group_array nesting pattern (§7.10, see `Registry\Queries`).
 * `quantity` is an `int32` column: PHP's `int` needs the token override.
 */
#[Table('transaction_details')]
#[Index(['transactionId'])]
final class TransactionDetail extends BaseModel
{
    #[Key]
    #[Generated]
    #[Column]
    public int $id;

    #[Column]
    #[ForeignKey(Transaction::class)]
    public int $transactionId;

    /** Populated only by the library (Level 2 loading); no public setter, so it can never disagree with `transactionId`. */
    #[ManyToOne('transactionId')]
    public private(set) ?Transaction $transaction = null;

    #[Column]
    public string $description;

    #[Column(type: ColumnType::Int32)]
    public int $quantity;

    #[Column]
    public Decimal $unitPrice;
}
